main.cpp

```diff
@@ -19,13 +19,26 @@ using namespace clang::tooling;
 
 class CastCallBack : public MatchFinder::MatchCallback {
 public:
-    CastCallBack(Rewriter& rewriter) {
-        // Your code goes here
-    };
-
+    CastCallBack(Rewriter& rewriter): _rewriter(rewriter) {};
     void run(const MatchFinder::MatchResult &Result) override {
-        // Your code goes here
+        const auto *cast_expr = Result.Nodes.getNodeAs<CStyleCastExpr>("cast");
+        auto replace_range = CharSourceRange::getCharRange(cast_expr->getLParenLoc(),

```

Comment:
What is the difference between `getCharRange()` and `getTokenRange()`?

Reply:
```getCharRange()``` returns the range of tokens before the start of the last token.
```getTokenRange()``` returns the range of characters up to and including the last character.

Reply:
Accepted.
To see the difference between these two methods, you can run the following code:
  
```cpp
  // Difference between getCharRange and getTokenRange 
  auto charRange = Lexer::getAsCharRange(
      CharSourceRange::getCharRange(node->getLParenLoc().getLocWithOffset(1),
      node->getRParenLoc().getLocWithOffset(-2)),
      SM, Result.Context->getLangOpts());
  
  auto tokenRange = Lexer::getAsCharRange(
      CharSourceRange::getTokenRange(node->getLParenLoc().getLocWithOffset(1),
      node->getRParenLoc().getLocWithOffset(-2)),
      SM, Result.Context->getLangOpts());
  
  llvm::outs() << "\n char range start: " << charRange.getBegin().printToString(SM) << ",\n char range end: " << charRange.getEnd().printToString(SM) << "\n";
  llvm::outs() << "\n token range start: " << tokenRange.getBegin().printToString(SM) << ",\n token range
 end: " << tokenRange.getEnd().printToString(SM) << "\n";
```  
  
  The output will be as follows:
  
```bash
   char range start: /UNN-C-Style-Checker/test/test.cpp:3:14,
   char range end: /UNN-C-Style-Checker/test/test.cpp:3:15
  
   token range start: /UNN-C-Style-Checker/test/test.cpp
:3:14,
   token range end: /UNN-C-Style-Checker/test/test.cpp:3:17
```  
  For the following example:
 ```cpp
  int main() {
      float foo;
      int i = (int)     foo ;
            // i - 15th item in line
            //   t - 17th item in line  
      return 0;
  }
  ```